system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj

```diff
@@ -349,8 +349,32 @@
                  (= expected-has-granules actual-has-granules))
                :xml (search/find-refs :collection {:include-has-granules true :include-granule-counts true})
                :echo10 (search/find-metadata :collection :echo10 {:include-has-granules true :include-granule-counts true})
+               :iso19115 (search/find-metadata :collection :iso19115 {:include-has-granules true :include-granule-counts true})
                :atom (search/find-concepts-atom :collection {:include-has-granules true :include-granule-counts true})
-               :atom (search/find-concepts-json :collection {:include-has-granules true :include-granule-counts true})))))))
+               :atom (search/find-concepts-json :collection {:include-has-granules true :include-granule-counts true}))))
+
+      (testing "granule_count"
+        (are3 [result-format results]
+          (let [expected-granule-count (util/map-keys :concept-id {coll1 2 coll2 2})
+                actual-granule-count (gran-counts/results->actual-granule-count result-format results)]
+            (= expected-granule-count actual-granule-count))
+          "granule count in xml format"
+          :xml (search/find-refs :collection {:include-has-granules true :include-granule-counts true})
+
+          "granule count in echo10 format"
+          :echo10 (search/find-metadata :collection :echo10 {:include-has-granules true :include-granule-counts true})
+
+          "granule count in iso format"
+          :iso19115 (search/find-metadata :collection :iso19115 {:include-has-granules true :include-granule-counts true})
+
+          "granule count in umm_json format"
+          :umm_json (search/find-concepts-umm-json :collection {:include-has-granules true :include-granule-counts true})
+
+          "granule count in atom format"
+          :atom (search/find-concepts-atom :collection {:include-has-granules true :include-granule-counts true})
+
+          "granule count in json format"
+          :atom (search/find-concepts-json :collection {:include-has-granules true :include-granule-counts true}))))))
 
```

Comment:
Can you add a test that you get the appropriate error message for both opendata and kml requests?

Reply:
I was just working on adding the kml one. Didn't think of opendata, will add that too. Thanks.

Reply:
opendata format does throw 500 error, even though the ticket didn't complain about this since it's not supported. Added the proper handling and the testing for it.